tests/mr/device/mr_test.hpp

```diff
@@ -234,6 +234,10 @@ struct mr_test : public ::testing::TestWithParam<mr_factory> {
   {
     auto factory = GetParam().factory;
     mr           = factory();
+    if (mr == nullptr) {
+      GTEST_SKIP() << "Skipping tests since the memory resource is not supported with this CUDA "
```

Comment:
Putting the `GTEST_SKIP` directly in the `make_cuda_async` doesn't work? I'm not sure what `GTEST_SKIP` actually does or what scope it needs to be used in.

Reply:
It's a macro that only works in a test method or `SetUp`.

Reply:
Ah, I thought it might throw an exception that gtest knows to catch or something.